pkg/provider/azure_vmss.go

```diff
@@ -210,7 +211,7 @@ func (ss *scaleSet) getVmssVM(nodeName string, crt azcache.AzureCacheReadType) (
 
 // GetPowerStatusByNodeName returns the power state of the specified node.
 func (ss *scaleSet) GetPowerStatusByNodeName(name string) (powerState string, err error) {
-	managedByAS, err := ss.isNodeManagedByAvailabilitySet(name, azcache.CacheReadTypeUnsafe)
```

Comment:
This change would cause every not found node to trigger cache updates and may cause issues when a large number of nodes are deleted or newly created. Have you done some validations on these scenarios?

Reply:
Correct. See my note in PR description: 
> Note that this approach minimizes cache refreshes by not refreshing unless necessary, but it does add extra GET API calls whenever a new node joins the cluster. This can be skipped altogether with the DisableAvailabilitySetNodes flag that is already implemented. It's not perfect but it will fix most scenarios for CAPZ. In the future, I'd like to propose a new "mixed" or "hybrid" VMSet implementation that is truly aware of both VMs and VMSS nodes as a more long-term fix.

We already do the same thing for VMSS: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure_vmss.go#L182-L188

There isn't a way to avoid this, unfortunately, if the node just joined the cluster and we've never fetched its corresponding VM/VMSS.

Reply:
since we are doing the refresh below based on the new diff between new nodes and known vms, I am not sure that this change is required?  If it isn't in the cache but the node is new then it is forced by https://github.com/kubernetes-sigs/cloud-provider-azure/pull/537/files#diff-e6715e02b5aa5f8b693bf3e14dd99dd9a160f0f0f1481df96891af24fc3a5fffR333

Reply:
> may cause issues when a large number of nodes are deleted or newly created

This will only be an issue if the nodes come up in sequence? I.E. if a node is added every 5 mins we would have an api call every five minutes.  

Do we have any data on how often that type scenario occurs?  Maybe during node auto scaler operations?

Reply:
Yeah, I think we can keep the cache read type unsafe as we are doing a force refresh if not found in the cache.

Reply:
> Yeah, I think we can keep the cache read type unsafe as we are doing a force refresh if not found in the cache.

Correct, reverted that change

Reply:
thanks for the discussion and addressing the issues